src/editor-sidebar/json-editor-modal.js

```diff
@@ -0,0 +1,37 @@
+import { useState, useEffect } from '@wordpress/element';
+import { Modal } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import CodeMirror from '@uiw/react-codemirror';
+import { json } from '@codemirror/lang-json';
+import { fetchThemeJson } from '../resolvers';
+
+const ThemeJsonEditorModal = ( { onRequestClose } ) => {
+	const [ themeData, setThemeData ] = useState( '' );
+	const themeName = useSelect( ( select ) =>
+		select( 'core' ).getCurrentTheme()
+	)?.name?.raw;
+	const fetchThemeData = async () => {
+		setThemeData( await fetchThemeJson() );
+	};
+	const handleSave = () => {};
+
+	useEffect( () => {
+		fetchThemeData();
+	} );
+
+	return (
+		<Modal
+			isFullScreen
+			title={ `theme.json for ${ themeName }` }
+			onRequestClose={ onRequestClose }
+		>
+			<CodeMirror
+				extensions={ [ json() ] }
+				value={ themeData }
+				onChange={ handleSave }
```

Comment:
```suggestion
				onChange={ handleSave }
				readOnly={ true }
```

I think we should add this `readOnly` option until we've implemented the save functionality. It's a little confusing that you're able to edit the JSON content - users might assume it's being saved automagically. The `readOnly` option disables any editing within the CodeMirror.